sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,72 @@
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::{Error, BoxDynError};
+use crate::postgres::{PgArgumentBuffer, PgValueFormat, PgTypeInfo, PgValueRef, Postgres};
+use crate::types::Type;
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>
+}
+
+impl PgLTree {
+    pub fn push(&mut self, label: String) -> Result<(), Error> {
+        if label.chars().all(|c| c.is_ascii_alphabetic() || c.is_ascii_digit() || c == '_') {
+            self.labels.push(label);
+            Ok(())
+        } else {
+            Err(Error::InvalidLtreeLabel)
+        }
+    }
+}
+
+impl FromStr for PgLTree {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        Ok(
+            Self {
+                labels: s.split('.').map(|s| s.to_owned()).collect()
+            }
+        )
+    }
+}
+
+impl Display for PgLTree {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.labels.join("."))
```

Comment:
This seems inefficient to me, as it allocates just to write to `f`. I'd go with a manual implementation